src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RazorComponentRenameEndpoint.cs

```diff
@@ -0,0 +1,292 @@
+                return documentSnapshot;
+            }, cancellationToken, TaskCreationOptions.None, _foregroundDispatcher.ForegroundScheduler).ConfigureAwait(false);
+
+            if (documentSnapshot is null)
+            {
+                return;
+            }
+
+            var codeDocument = await documentSnapshot.GetGeneratedOutputAsync().ConfigureAwait(false);
+            if (codeDocument.IsUnsupported())
+            {
+                return;
+            }
+
+            if (!FileKinds.IsComponent(codeDocument.GetFileKind()))
+            {
+                return;
+            }
+
+            var uri = new UriBuilder
+            {
+                Path = documentPath,
+                Host = string.Empty,
+                Scheme = Uri.UriSchemeFile,
+            }.Uri;
+            AddEditsForCodeDocument(documentChanges, originTagHelperBinding, newName, uri, codeDocument);
+        }
+
+        public void AddEditsForCodeDocument(List<WorkspaceEditDocumentChange> documentChanges, TagHelperBinding originTagHelperBinding, string newName, Uri uri, RazorCodeDocument codeDocument)
+        {
+            var documentIdentifier = new VersionedTextDocumentIdentifier { Uri = uri };
+            foreach (var node in codeDocument.GetSyntaxTree().Root.DescendantNodes().Where(n => n.Kind == SyntaxKind.MarkupTagHelperElement))
+            {
+                if (node is MarkupTagHelperElementSyntax tagHelperElement && BindingsMatch(originTagHelperBinding, tagHelperElement.TagHelperInfo.BindingResult))
+                {
+                    documentChanges.Add(new WorkspaceEditDocumentChange(new TextDocumentEdit
+                    {
+                        TextDocument = documentIdentifier,
+                        Edits = CreateEditsForMarkupTagHelperElement(tagHelperElement, codeDocument, newName)
+                    }));
+                }
+            }
+        }
+
+        public List<TextEdit> CreateEditsForMarkupTagHelperElement(MarkupTagHelperElementSyntax element, RazorCodeDocument codeDocument, string newName)
+        {
+            var edits = new List<TextEdit>
+            {
+                new TextEdit()
+                {
+                    Range = element.StartTag.Name.GetRange(codeDocument.Source),
+                    NewText = newName,
+                },
+            };
+            if (element.EndTag != null)
+            {
+                edits.Add(new TextEdit()
+                {
+                    Range = element.EndTag.Name.GetRange(codeDocument.Source),
+                    NewText = newName,
+                });
+            }
+            return edits;
+        }
+
+        private static bool BindingsMatch(TagHelperBinding left, TagHelperBinding right)
+        {
+            foreach (var leftDescriptor in left.Descriptors)
+            {
+                foreach (var rightDescriptor in right.Descriptors)
+                {
+                    if (leftDescriptor.Equals(rightDescriptor))
```

Comment:
Is there a reason for this over just `==`?
```suggestion
                    if (leftDescriptor == rightDescriptor)
```

Reply:
Yes. For string equality it is always advised to use string.Equals and supply a StringComparison(Ordinal unless you have specific reasons for doing otherwise). Without this behaviors can differ when run on different machines due to different cultures (as determined by machine cultures) handling equality in different ways.
```suggestion
                    if (leftDescriptor.Equals(rightDescriptor, StringComparison.Ordinal))
```